src/nyc_trees/apps/home/training/types.py

```diff
@@ -0,0 +1,107 @@
+# -*- coding: utf-8 -*-
+from __future__ import print_function
+from __future__ import unicode_literals
+from __future__ import division
+
+from django.core.urlresolvers import reverse
+from django.contrib.auth.decorators import login_required
+from django_tinsel.decorators import route
+from django_tinsel.utils import decorate as do
+from apps.home.training.decorators import render_flatpage, mark_user
+from django.contrib.flatpages.views import flatpage
+
+_PURE_TEMPLATE = '%s_pure'
+_MARK_PROGRESS_TEMPLATE = '%s_mark_progress'
+_TRAINING_FINISHED_TEMPLATE = 'training_finished_%s'
+
+
+class TrainingGateway(object):
```

Comment:
This class has some useful methods, but I don't see any value in trying to make it "look like" a training step. It just confused me.

Consider making `get_step`, `is_complete`, `get_context`, and `get_steps` module-level functions and ditching `TrainingGateway`.